Add grpc ingestion test schedule to http-logs

finnegancarroll opened the pull request:
### Description
- Migrates the gRPC index append operation to common operations in workload setup.
- Adds gRPC index append test procedure to `http_logs` workload.
- Adds version guards for gRPC index append

Version guards are optimistic. When no distribution-version is provided we will still attempt the gRPC index append operation for whatever workload is being run.

When distribution version is in compatible benchmarks exit and error is written to `benchmark.log`:
```
jinja2.exceptions.TemplateNotFound: error: Distribution version is 1.1.1 - gRPC is not supported in versions < 3.3.0
```



### Issues Resolved
N/A

### Testing
~~- [] New functionality includes testing~~

Manual testing with OSB for both http_logs and big5 workloads.
OSB command format.
```
export HOST=localhost
opensearch-benchmark run --kill-running-processes \
    --pipeline=benchmark-only \
    --target-hosts=${HOST}:9200 \
    --grpc-target-hosts=${HOST}:9400 \
    --workload-path "<path>/opensearch-benchmark-workloads/http_logs" \
    --test-procedure="grpc-append-no-conflicts-index-only" \
    --workload-params '{"number_of_shards":"1","number_of_replicas":"0", "ingest_percentage":"0.1", "distribution_version":"2.14.0"}'
```

Test plan
- No distribution workload param
Workloads pass
- Greater than 3.3.0 distribution workload param
Workloads pass
- Explicit <3.3.0 workload param
Workloads fail with error in logs

Testing new http_logs gRPC workload:
```
[INFO] [Test Run ID]: d08fe186-179b-400b-9a48-4ebd79191f75
[INFO] Running test with workload [http_logs], test_procedure [grpc-append-index-only] and cluster_config ['external'] with version [3.4.0-SNAPSHOT].

Running delete-index                                                           [100% done]
Running create-index                                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running grpc-index-append                                                      [100% done]
Running refresh-after-index                                                    [100% done]
Running force-merge                                                            [100% done]
Running refresh-after-force-merge                                              [100% done]
Running wait-until-merges-finish                                               [100% done]
...
----------------------------------
[INFO] ✅ SUCCESS (took 67 seconds)
----------------------------------
```

[Describe how this change was tested]

### Backport to Branches:
- [ ] 6
- [ ] 7
- [ ] 1
- [ ] 2
- [x] 3

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

IanHoang commented on http_logs/test_procedures/grpc/grpc-schedule.json:
```diff
@@ -0,0 +1,56 @@
+    }
+},
+{
+    "name": "check-cluster-health",
```
We should embed the common_operations like this https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/3d53dae7096eeb3c281c3ef15e3744adcb3d60f2/http_logs/test_procedures/default.json#L6-L8

finnegancarroll commented:
Hi @IanHoang , thanks for the feedback. I did a refactor of the changes according to your two points:
- Add version guard for gRPC
- Make gRPC indexing a common operation.

I've moved gRPC index append to /common_operations and migrated http_logs/big5 to use grpc_workload_setup.json instead of including this logic directly in the test procedures. This common operation additionally has an added version guard for >= 3.3.0.

One note is the big5 workload already has some version guards and sets a default distribution version of 2.11.0 in a couple places. I added a `  {% set unknown_distribution = "True" %}` to the top level `big5/workload.json` to allow gRPC to execute optimistically if the distribution is unknown without impacting the existing default version of 2.11.0.

Please take another look when you have a chance. Thanks!
